eos-updater-prepare-volume/eos-updater-prepare-volume

```diff
@@ -173,6 +176,13 @@ class VolumePreparer:
 
         return runtimes
 
+    def _add_autoinstall_flatpaks(self, flatpak_collection_refs):
+        applied_actions = EosUpdaterFlatpakInstaller.determine_flatpak_ref_actions_to_check(None)
```

Comment:
Not entirely sure this is the right function to call. `determine_flatpak_ref_actions_to_check()` will load the autoinstall lists, parse them, and then *filter out* any entries whose serial number is lower than the latest applied serial number for this machine. I think you probably want to take part of the implementation for `determine_flatpak_ref_actions_to_check()` and call that instead. i.e. Call `flatpak_ref_actions_and_progresses()` and not `euu_filter_for_existing_flatpak_ref_actions()`. Since `flatpak_ref_actions_and_progresses()` is not public, it’s probably best to just copy part of its implementation (the bit relating to `out_actions`) instead. i.e. Call `euu_flatpak_ref_actions_from_paths()`.

Also, the above paragraph aside, did you consider calling `euu_flatten_flatpak_ref_actions_table()` on the `applied_actions` and reject the idea on the basis that the computer the updates are being applied to might have a different set of autoinstall list overrides, and hence it would be better to copy apps onto the USB stick even if their autoinstall list entries were overridden (squashed) on this machine?

Reply:
In light of what @pwithnall here says about the overrides being potentially different on the target, the best bet here is to read in all the actions that might be necessary on the ostree and squash them from there.

I'm not sure how much can be done to handle the overrides situation - the source cant know what ref actions will be needed on the target since /var and /etc are not really managed by ostree